### DISCUSSION TITLE
I want to  implement autoscroll

### DISCUSSION THREAD

**Discussion by Artemis-9** [patrykandpatrick/vico#585] [Questions]
I'm new to jetpack compose. I have an implementation of line chart for 3 weeks. i just want to show the line chart for current week with values from Mon to Sat. i have a button for prev and next week. on clicking the button i want to auto scroll to next week . can anyone suggest a way to implement this functionality.

**Comment by Laul** [comment]
I've quite the same question, but in addition to your description I 'd like to trigger it from another graph.  ( #574)
I dont think we can use the autoscroll, because it only implements start and end options. 
I was considering having a dynamic way to consider the data based on the x value, but a little tricky in my opinion 😅

**Comment by patrickmichalik** [comment]
Hello! Here, you’ll need programmatic scrolling. [Vico 2.0.0 Alpha 9](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.9) introduces the required features. First, you’ll most likely want to set `scrollEnabled` (in `rememberVicoScrollState`) and `zoomEnabled` (in `rememberVicoZoomState`) to `false`. To show seven days’ worth of data, set `initialZoom` (in `rememberVicoZoomState`) to `Zoom.x(7f)`. To scroll forward by a week, use the following. To scroll the other way, replace `7f` with `-7f`. `scrollState` is a reference to the chart’s `VicoScrollState`.

```kt
scrollState.animateScroll(Scroll.Relative.x(7f))
```
